File: Swift/buddy-strings.swift
class Solution {
    func buddyStrings(_ A: String, _ B: String) -> Bool {
        guard A.count == B.count else {
            return false
        }
        var diff = [(Character, Character)]()

        for (a, b) in zip(A, B) {
            if a != b {
                diff.append((a, b))
                if diff.count > 2 {
                    return false
                }
            }
        }

        return (diff.count == 0 && Set(Array(A)).count < A.count) ||
                (diff.count == 2 && diff[0] == (diff[1].1, diff[1].0))
    }
}

File: Python/buddy-strings.py
class Solution:
    def buddyStrings(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: bool
        """
        if len(A) != len(B):
            return False

        diff = []

        for a, b in zip(A, B):
            if a != b:
                diff.append((a, b))
                if len(diff) > 2:
                    return False

        return (not diff and len(set(A)) < len(A)) or \
               (len(diff) == 2 and diff[0] == diff[1][::-1])
